### ISSUE TITLE
CKM factor in K -> pi nu nu amplitude

### ISSUE THREAD

**Issue by girishky** [flav-io/flavio#185]
Hi,
Flavio basis convention (https://wcxf.github.io/assets/pdf/WET-3.flavio.pdf) for s -> d nunu transition  has CKM factor (Vts* Vtd) in operators' definition. But in flavio implementation, amplitude for K -> pi nu nu has `xi('t', 'sd')` which is basically (Vts Vtd*)
https://github.com/flav-io/flavio/blob/3d11de09d25507f093cce1ad6e3d970a379baf88/flavio/physics/kdecays/kpinunu.py#L11

Shouldn't it be `xi('t', 'ds')` in the above to be consistent with flavio operator definition?

**Comment by DavidMStraub** [comment]
In the observables, only the absolute value or the square of the imaginary part of the amplitude enters. so it shouldn't matter whether it's the d->s or the s->d amplitude. Do you agree?

**Comment by girishky** [comment]
Hi David, Thanks for reply. I agree with what you said. But my point is that in K -> pi nunu amplitude function, `X` is basically WC for s -> d transition https://github.com/flav-io/flavio/blob/3d11de09d25507f093cce1ad6e3d970a379baf88/flavio/physics/kdecays/kpinunu.py#L20

but the CKM factor that is multiplied by WC here https://github.com/flav-io/flavio/blob/3d11de09d25507f093cce1ad6e3d970a379baf88/flavio/physics/kdecays/kpinunu.py#L24 is related to conjugate (d->s) transition.

For absolute value, I agree this won't matter. But imaginary part  of amplitude (even square of it) would be different if `X` also contain complex phase due to NP. So shouldn't amplitude be `amp = xi_t.conj() * X` or `amp = xi_t * X.conj()`  ?

**Comment by hoodyn** [comment]
I agree with @girishky. Simply taking imaginary part for different neutrino species is inappropriate. It is the same kind of error that has been reported in #138 for K0->ll'.

**Comment by DavidMStraub** [comment]
But that's a different issue...

**Comment by peterstangl** [comment]
> In the observables, only the absolute value or the square of the imaginary part of the amplitude enters. so it shouldn't matter whether it's the d->s or the s->d amplitude. Do you agree?

I agree with this as long as in
https://github.com/flav-io/flavio/blob/3d11de09d25507f093cce1ad6e3d970a379baf88/flavio/physics/kdecays/kpinunu.py#L24

the `X` is strictly real. However, I agree with @girishky that if `X` also has an imaginary part, then it *does* matter whether `xi('t', 'sd')` or `xi('t', 'ds')` is used.

I suggest to simply replace
https://github.com/flav-io/flavio/blob/3d11de09d25507f093cce1ad6e3d970a379baf88/flavio/physics/kdecays/kpinunu.py#L11-L12
by
```python
 xi_t = flavio.physics.ckm.xi('t', 'ds')(par) 
 xi_c = flavio.physics.ckm.xi('c', 'ds')(par) 
 ```
 @DavidMStraub @girishky do you both agree with this solution?

**Comment by girishky** [comment]
@peterstangl  Yes. This is better. Thanks.

best,

**Comment by peterstangl** [comment]
This issue has been fixed in PR #189.
